coalib/parsing/Globbing.py

```diff
@@ -358,49 +367,51 @@ def _iglob(pattern):
         relative_glob_function = relative_flat_glob
 
     if not dirname:
-        for file in relative_glob_function(dirname, basename):
+        for file in relative_glob_function(dirname, basename, ignored_globs):
             yield file
         return
 
     # Prevent an infinite recursion if a drive or UNC path contains
     # wildcard characters (i.e. r'\\?\C:').
     if dirname != pattern and has_wildcard(dirname):
-        dirs = iglob(dirname)
+        dirs = iglob(dirname, ignored_globs)
     else:
         dirs = [dirname]
 
     for dirname in dirs:
-        for name in relative_glob_function(dirname, basename):
+        for name in relative_glob_function(dirname, basename, ignored_globs):
             yield os.path.join(dirname, name)
 
 
 @yield_once
-def iglob(pattern):
+def iglob(pattern, ignored_globs=None):
     """
     Iterates all filesystem paths that get matched by the glob pattern.
     Syntax is equal to that of fnmatch.
 
-    :param pattern: Glob pattern with wildcards
-    :return:        Iterator that yields all file names that match pattern
+    :param pattern:       Glob pattern with wildcards
+    :param ignored_globs: List of globs to ignore when matching files
+    :return:              Iterator that yields all file names that match pattern
```

Comment:
I already noticed this, and I think we can make a new issue for that. Let's improve speed for now, and increase consistency in a separate PR, so we get something merged.